service/forward_service.cc

```diff
@@ -200,21 +379,28 @@ future<query::forward_result> forward_service::dispatch_to_shards(
     std::optional<tracing::trace_info> tr_info
 ) {
     _stats.requests_dispatched_to_own_shards += 1;
+    std::optional<query::forward_result> result;
+    std::vector<future<query::forward_result>> futures;
 
-    auto result = co_await container().map_reduce0(
-        [req, tr_info] (auto& fs) {
+    for (const auto& s : smp::all_cpus()) {
+        futures.push_back(container().invoke_on(s, [req, tr_info] (auto& fs) {
```

Comment:
Why the change from map_reduce?

Reply:
because of the `with_thread_if_needed` required and reduce function in `map_reduce0` cannot return future